raiden/channel/netting_channel.py

```diff
@@ -594,13 +582,30 @@ def register_transfer_from_to(
         amount = transfer.transferred_amount - from_state.transferred_amount
         distributable = from_state.distributable(to_state)
 
-        if amount > distributable:
-            raise InsufficientBalance(transfer)
+        if isinstance(transfer, DirectTransfer):
+            if amount > distributable:
+                raise InsufficientBalance(transfer)
 
-        if isinstance(transfer, LockedTransfer):
+        elif isinstance(transfer, LockedTransfer):
             if amount + transfer.lock.amount > distributable:
                 raise InsufficientBalance(transfer)
 
+        elif isinstance(transfer, Secret):
+            hashlock = sha3(transfer.secret)
+            lock = to_state.balance_proof.get_lock_by_hashlock(hashlock)
+            transferred_amount = from_state.transferred_amount + lock.amount
+
+            # transfer.transferred_amount could be larger than the previous
+            # transferred_amount + lock.amount, that scenario is a bug of the
```

Comment:
`that scenario is a bug of the payer` -> `that scenario indicates a bug on the payer side`